src/test/scala/mesosphere/marathon/upgrade/DeploymentDeathRowTest.scala

```diff
@@ -0,0 +1,71 @@
+package mesosphere.marathon.upgrade
+
+import mesosphere.marathon.Protos.MarathonTask
+import mesosphere.marathon.state.Timestamp
+import mesosphere.marathon.tasks.MarathonTasks
+import org.scalatest.FunSuite
+
+class DeploymentDeathRowTest extends FunSuite {
```

Comment:
It is nice that your separated some of the logic to be testable in isolation.

But you also need a more high-level test that tests the changes in the DeploymentActor.